src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -84,8 +84,14 @@ class ReimbursementAccountPage extends React.Component {
         this.continue = this.continue.bind(this);
         this.getDefaultStateForField = this.getDefaultStateForField.bind(this);
         this.goBack = this.goBack.bind(this);
+
+        // The first time we open this page, the props.reimbursementAccount has not been loaded from the server.
+        // Calculating shouldShowContinueSetupButton on the default data doesn't make sense, and we should recalculate
+        // it once we get the response from the server the first time in componentDidUpdate.
+        const hasACHDataBeenLoaded = this.props.reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps;
```

Comment:
What if they are loaded and happen to be equal to the default props? How can we be sure to differentiate those two scenarios? I'm sure we have many similar issues throughout the entire App where we don't know if a given prop is loaded and is default or not loaded at all.

Reply:
Do you think we should hold for navigation reboot? Either way, I think it would be better to show a skeleton loader while the props are not loaded and the `ReimbursementAccountPage` shouldn't be rendered at all so we avoid the overhead `hasACHDataBeenLoaded` logic.

Reply:
> What if they are loaded and happen to be equal to the default props?

It is impossible that the loaded data will be equal by reference (`===`) to the default data.

> Do you think we should hold for navigation reboot? 

No, I don't see why, this has little to do with navigation in my opinion.

> Either way, I think it would be better to show a skeleton loader while the props are not loaded and the ReimbursementAccountPage shouldn't be rendered at all so we avoid the overhead hasACHDataBeenLoaded logic.

About the skeleton, I think it could have a nicer effect, but this is more work and I consider it out of scope. There would be also some ambiguity about which skeleton view you show... you don't know in which state the bank account is, if it even exists.

I'm not really changing what we render with these changes... we were already showing a loader when `reimbursementAccount.isLoading = true`, and that has not changed. `this.hasACHDataBeenLoaded` is just to keep track of the case:

1. Login
2. Go to the VBBA for the first time -> we haven't loaded the data

because it is useful for knowing when we have to calculate the flag `shouldShowContinueSetupButton` that shows the Continue/Start over view.

Reply:
Thanks for your confirmation. In that case I think we can proceed with the tests. Please mark this as resolved.